src/_gettsim_tests/test_join.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+import pytest
+from _gettsim.shared import join_numpy
+
+
+@pytest.mark.parametrize(
+    "foreign_key, primary_key, target, expected",
+    [
+        (
+            np.array([1, 2, 3]),
+            np.array([1, 2, 3]),
+            np.array(["a", "b", "c"]),
+            np.array(["a", "b", "c"]),
+        ),
+        (
+            np.array([3, 2, 1]),
+            np.array([1, 2, 3]),
+            np.array(["a", "b", "c"]),
+            np.array(["c", "b", "a"]),
+        ),
+        (
+            np.array([1, 1, 1]),
+            np.array([1, 2, 3]),
+            np.array(["a", "b", "c"]),
+            np.array(["a", "a", "a"]),
+        ),
+    ],
+)
+def test_join_numpy(
+    foreign_key: np.ndarray[int],
+    primary_key: np.ndarray[int],
+    target: np.ndarray[str],
+    expected: np.ndarray[str],
+):
+    assert np.array_equal(join_numpy(foreign_key, primary_key, target), expected)
+
+
+def test_join_numpy_raises_duplicate_primary_key():
+    with pytest.raises(ValueError):
```

Comment:
Would be good to check the error message, also to differentiate from subsequent test.